src/common/history.c

```diff
@@ -1409,6 +1409,7 @@ GList *dt_history_duplicate(GList *hist)
     if(params_size > 0)
     {
       new->params = malloc(params_size);
+      // FIXME this is not doing a deep copy of params
```

Comment:
I'm not sure to understand, this is a deep copy as there is no reference in params just a flat/serialized set of values.

Reply:
See the comment I added in `useless.c`; it is suggested there that allocating arrays and referencing them from params is allowed, but it isn't (and currently doesn't happen anywhere).

Reply:
And that's good to know that it doesn't happen anywhere. This would break darktable in many places. I don't even see how this coud work in presets, styles? And even in the current way of storing params in XMP and DB. So let's pretend this is not supported at all.